Dockerfile

```diff
@@ -1,27 +1,18 @@
 FROM node:8 AS builder
 
-RUN useradd --create-home lisk && \
-    apt-get update && \
-    apt-get install -y nasm && \
-    npm install --global bower
-# As of Mai 2018 cloud.docker.com runs docker 17.06.1-ce
-# however v17.12 is required to use the chown flag
-#COPY --chown=lisk:lisk . /home/lisk/lisk-explorer/
-COPY . /home/lisk/lisk-explorer/
-RUN chown lisk:lisk --recursive /home/lisk/lisk-explorer
-
-USER lisk
+RUN useradd --create-home lisk
+COPY --chown=lisk:lisk ./package-lock.json ./package.json /home/lisk/lisk-explorer/
 WORKDIR /home/lisk/lisk-explorer
-RUN npm install && \
-    npm run build
+USER lisk
```

Comment:
What is the advantage of splitting (2 `COPY` and `RUN` statements) and reordering?

Reply:
in this case, the first part of COPY package.json and package-lock.json then RUN `npm ci`will create a layer with the dependencies, and this layer can be cached to be use in future builds, so you don't need to download and install everything again if the code changes. it will recreate again only if these files change.

Reply:
As discussed there is no way to guarantee the second `COPY` will not overwrite some files under `node_modules`; try adding this directory to `.dockerignore` maybe?

Reply:
it's already there! LOL https://github.com/LiskHQ/lisk-explorer/blob/2bfec41ae88fa92d535c064ffc50a92bf8be7c4e/.dockerignore#L6